safaribooks.py

```diff
@@ -899,17 +899,20 @@ def _thread_download_images(self, url):
         self.display.state(len(self.images), self.images_done_queue.qsize())
 
     def _start_multiprocessing(self, operation, full_queue):
-        if len(full_queue) > 5:
-            for i in range(0, len(full_queue), 5):
-                self._start_multiprocessing(operation, full_queue[i:i + 5])
-
-        else:
-            process_queue = [Process(target=operation, args=(arg,)) for arg in full_queue]
-            for proc in process_queue:
-                proc.start()
-
-            for proc in process_queue:
-                proc.join()
+            try:
+                if len(full_queue) > 5:
+                    for i in range(0, len(full_queue), 5):
+                        self._start_multiprocessing(operation, full_queue[i:i + 5])
+    
+                else:
+                    process_queue = [Process(target=operation, args=(arg,)) for arg in full_queue]
+                    for proc in process_queue:
+                        proc.start()
+    
+                    for proc in process_queue:
+                        proc.join()
+            except IOError as e:
+                self.display.error(f"IOError occurred: {e}")
```

Comment:
**Suggestion:** The IOError exception handling is too broad and may hide other issues. Consider specifying the exact error you're expecting, or add more specific error handling.
```suggestion
            try:
                ...
            except IOError as e:
                if e.errno == errno.ENOENT:
                    self.display.error(f"File not found error occurred: {e}")
                else:
                    raise e
```